### PR TITLE
Add name to --drivers

### PR DESCRIPTION
**Pull request by sunce86**
# Description
`--drivers` argument now requires for each driver to contain the name, besides url.
It's similar to how the `--price_estimation_drivers` argument is working.

This is needed for nicer debugging/metrics and also to populate the `SolverCompetitionDB` solver names.

### CODE REVIEW

File: crates/autopilot/src/arguments.rs [hunk 1]
```diff
@@ -153,9 +153,9 @@ pub struct Arguments {
     #[clap(long, env, action = clap::ArgAction::Set, default_value = "false")]
     pub enable_colocation: bool,
 
-    /// Driver base URLs.
+    /// A list of drivers in the following format: `<NAME>|<URL>,<NAME>|<URL>`
     #[clap(long, env, use_value_delimiter = true)]
-    pub drivers: Vec<Url>,
+    pub drivers: Vec<ExternalSolver>,
```
Comment (MartinquaXD) [inline crates/autopilot/src/arguments.rs:158]: nit: Either we don't consider Gnosis solvers external then this name doesn't capture all solvers. Or we consider them external then the `External` prefix is kind of redundant.
Maybe `DriverConfig`?

**Review by MartinquaXD — APPROVED** [review]

File: crates/autopilot/src/arguments.rs (hunk 1 above)
Comment (MartinquaXD) [inline crates/autopilot/src/arguments.rs:158]: nit: This should probably be renamed when we move entirely to colocated drivers.

Comment (sunce86) [reply crates/autopilot/src/arguments.rs:158]: Totally agreed.

**Review by nlordell — APPROVED** [review]
Thank you! This will help fix our metrics in the panel as well 😄.

**Review by fleupold — COMMENTED** [review]
Do we really need to name the driver? Don't we already have access to a short named **solver name** in the driver [here](https://github.com/cowprotocol/services/blob/d47691af72e19b1e5add4eb5965a841c9fef09b5/crates/driver/src/infra/solver/mod.rs#L75)? Why can't we use this one if the relationship between driver<>solver is 1:1?

E.g. I see short named solver in [this log](https://production-6de61f.kb.eu-central-1.aws.cloud.es.io/app/discover#/doc/ea511870-d9b3-11ed-a9d0-a17451f01cc1/cowlogs-staging-2023.10.12?id=4kQYIosBGP6wkTcj6I6z).

Can't we just add this name to the `Solved` struct we are returning (technically the solution is coming from the solver)? This way we don't need to make the configuration more involved.

**Comment by MartinquaXD** [comment]
> Can't we just add this name to the Solved struct we are returning (technically the solution is coming from the solver)?

That way solvers could report a bogous name and mess with out metrics. I don't expect this to be an attack vector but I would argue against baking in trust assumptions that don't have to be there.

**Comment by nlordell** [comment]
> Can't we just add this name to the Solved struct we are returning (technically the solution is coming from the solver)? This way we don't need to make the configuration more involved.

What happens (in the context of an external solver running a driver) when two solvers have the same name?

I view this "name" as an internal tag (maybe we can even call it that) that we use for metrics.

That being said, in the context of the `CompetitionDB` storing, it _may_ make sense for solvers to self-specify a descriptive name

**Comment by sunce86** [comment]
Do we have a consensus to merge this PR? 
I feel safer to use driver configuration for this after all.

**Comment by nlordell** [comment]
Personally, I think it should be merged. Lets see what @fleupold thinks.

**Comment by fleupold** [comment]
I'm fine either way. One argument for letting drivers self specify the name could be that at some point in the future the autopilot should just broadcast auctions (and drivers report solutions permissionlessly) in which case it may not know a list of all solvers. But that's still far in the future...

**Comment by nlordell** [comment]
> One argument for letting drivers self specify the name could be that at some point in the future the autopilot should just broadcast auctions (and drivers report solutions permissionlessly) in which case it may not know a list of all solvers.

I agree, but since we are pushing auctions to drivers at the moment, the `name` makes sense for now so that we have a label tying a driver to an auction.

**Pull request merged**